### ISSUE TITLE
🐛 False Positive in noPrivateImports Rule with Overlapping Type and Variable Exports

### ISSUE THREAD

**Issue by gediminas-braskus** [biomejs/biome#7481]
### Environment information

```block
CLI:
  Version:                      2.2.3
  Color support:                true

Platform:
  CPU Architecture:             aarch64
  OS:                           macos

Environment:
  BIOME_LOG_PATH:               unset
  BIOME_LOG_PREFIX_NAME:        unset
  BIOME_CONFIG_PATH:            unset
  BIOME_THREADS:                unset
  NO_COLOR:                     unset
  TERM:                         xterm-256color
  JS_RUNTIME_VERSION:           v24.6.0
  JS_RUNTIME_NAME:              node
  NODE_PACKAGE_MANAGER:         pnpm/10.15.1

Biome Configuration:
  Status:                       Loaded successfully
  Path:                         biome.jsonc
  Formatter enabled:            true
  Linter enabled:               true
  Assist enabled:               true
  VCS enabled:                  true

Workspace:
  Open Documents:               0
```

### What happened?

**Description**

When enabling the lint/correctness/noPrivateImports rule, an error occurs if both a type and a variable are exported with the same name, and both are annotated with the JSDoc tag /** @public */.

Biome throws the following error:
  _You may not import a symbol with package visibility from here._

**Steps to Reproduce**

1. Define a type and a variable with the same name.

2. Export both with /** @public */.

3. Import either the type or the variable in another file.

```
// types.ts

/** @public */
export const Example = {
  Example: 'example',
  Example1: 'example_1',
  Example2: 'example_2'
} as const

/** @public */
export type Example = (typeof Example)[keyof typeof Example]
```

```
// main.ts

import type { Example } from './types.ts'
```

### Expected result

Since both the type and the variable are explicitly marked as @public, importing either should not trigger a noPrivateImports violation.

### Code of Conduct

- [x] I agree to follow Biome's Code of Conduct

**Comment by ematipico** [comment]
Can you share a playground link? You can create multiple files and you should be able to trigger the rule

**Comment by biome-cookie** [comment]
Hello @gediminas-braskus, please provide a minimal reproduction. You can use one of the following options:

- Provide a link to [our playground](https://biomejs.dev/playground), if it's applicable.
- Provide a link to GitHub repository. To easily create a reproduction, you can use our interactive CLI via `npm create @biomejs/biome-reproduction`

Issues marked with `S-Needs repro` will be **closed** if they have **no activity within 3 days**.

**Comment by biome-cookie** [comment]
This issue has been automatically closed because it has been inactive for more than 3 days and it does not have a minimal reproduction.
